src/fclone.js

```diff
@@ -1,5 +1,13 @@
 'use strict';
 
+// see if it looks and smells like an iterable object, and do accept length === 0
+function isArrayLike(item) {
+  if (Array.isArray(item)) return true;
+
+  var len = item && item.length;
+  return typeof len === 'number' && (len === 0 || (len - 1) in item);
```

Comment:
`{length: 3, 0: 'test', 1: 'test', 2: 'test'}` however will be cloned as an array